src/components/RouteForm.js

```diff
@@ -0,0 +1,156 @@
+        });
+    };
+
+    const addOrUpdateRoute = () => {
+        if (isEditing) {
+            updateRoute();
+        } else {
+            addRoute();
+        }
+        onClose();
+    };
+
+    const addRoute = () => {
+        dispatch({
+            type: actionTypes.ADD_ROUTE,
+            payload: { default: is_default, destination, gateway, interface: device, options }
+        });
+    };
+
+    const isInComplete = () => {
+        if (!is_default && destination.length == 0) return true;
+        if (gateway.length == 0) return true;
+
+        return false;
+    };
+
+    return (
+        <Modal
+            variant={ModalVariant.small}
+            title={isEditing ? _("Edit Route") : _("Add Route")}
+            isOpen={isOpen}
+            actions={[
+                <Button key="confirm" variant="primary" onClick={addOrUpdateRoute} isDisabled={isInComplete()}>
+                    {isEditing ? _("Change") : _("Add")}
+                </Button>,
+                <Button key="cancel" variant="link" onClick={onClose}>
+                    {_("Cancel")}
+                </Button>
+            ]}
+        >
+            <Form>
+                <FormGroup
+                    label={_("Default route")}
+                    fieldId="is_default"
+                >
+                    <Checkbox
+                        id="is_default"
+                        isChecked={is_default}
+                        onChange={setDefault}
+                    />
+                </FormGroup>
+
+                { !is_default &&
+                    <FormGroup
+                        label={_("Destination")}
+                        isRequired={!is_default}
+                        fieldId="destination"
+                        helperText={_("Destination")}
+                    >
+                        <TextInput
+                            isRequired
+                            id="destination"
+                            value={destination}
+                            onChange={setDestination}
+                        />
+                    </FormGroup>}
+
+                <FormGroup
+                    label={_("Gateway")}
+                    isRequired
+                    fieldId="gateway"
+                >
+                    <TextInput
+                        isRequired
+                        id="gateway"
+                        value={gateway}
+                        onChange={setGateway}
+                    />
+                </FormGroup>
+
+                <FormGroup
+                    label={_("Device")}
+                    isRequired
+                    fieldId="device"
+                >
+                    <FormSelect value={device} onChange={setDevice} id="device">
+                        {candidateInterfaces.map(({ name }, index) => (
+                            <FormSelectOption key={index} value={name} label={name} />
```

Comment:
Just a note. According to an [article](https://medium.com/javascript-in-plain-english/you-wish-you-knew-this-little-reactjs-quirk-f021d8375dbe) that I read yesterday, maybe we should use the interface `id` as key instead the index.